### ISSUE TITLE
bug: Webhooks.Url is incorrect when windmill is located in non-root path

### ISSUE THREAD

**Issue by bigcat88** [windmill-labs/windmill#4190]
### Describe the bug

https://github.com/windmill-labs/windmill/blob/1c838767751e8c3a00c4536a5efdaf1b741853a2/frontend/src/lib/components/details/WebhooksPanel.svelte#L43-L74

`$page.url.origin` will always point to the root, and when the windmill is in a subpath(including backend's `/api`) - this subpath will be absent.

### To reproduce

_it is clear from code_

### Expected behavior

Currently I see three options how this can be fixed:

1. `BASE_URL` can be appended.
2. `VITE_BASE_URL` can be appended.
3. New environment variable when defined can be appended.

```js
let base = `${$page.url.origin}${$SOMETHING}/api/w/${$workspaceStore}/jobs`
```

### Browser information

not important

### Application version

v1.375.0

**Comment by marcelklehr** [comment]
After merging our PR the following two menu items still do not work, without us being able to figure out why:

* Schedules
* Audit Logs

**Comment by rubenfiszel** [comment]
@bigcat88 base from src/lib/base.ts derived from VITE_BASE_URL need to be appended

@marcelklehr they do not work as in they lead to the wrong page?

**Comment by marcelklehr** [comment]
> they do not work as in they lead to the wrong page?

Yes

**Comment by rubenfiszel** [comment]
@marcelklehr could you give me more details please so we can investigate. What is your Vite_base_url and what is the URL that those menu items led you to ?

**Comment by marcelklehr** [comment]
Could this be the fix? https://github.com/windmill-labs/windmill/pull/4205

**Comment by rubenfiszel** [comment]
no but i'm on it

**Comment by rubenfiszel** [comment]
both issues are fixed on latest @marcelklehr @bigcat88

**Comment by marcelklehr** [comment]
Cool. thank you so much!

**Comment by bigcat88** [comment]
Apologies for reaching out again, but we’ve identified another place where VITE_BASE_URL isn’t being appended.

This occurs when clicking on the "Flows" section and then navigating to the "Webhooks" tab on the bottom-right:

<img width="475" alt="image" src="https://github.com/user-attachments/assets/7c51c5c3-e449-4564-922b-e2af977f38a6">

Would it be possible to address this in a future update?

Thank you!

**Comment by rubenfiszel** [comment]
@bigcat88 fixed in very latest commit [776e978](https://github.com/windmill-labs/windmill/commit/776e97875218476a46bfb72f8648b83471e70cc4)

**Comment by bigcat88** [comment]
> @bigcat88 fixed in very latest commit [776e978](https://github.com/windmill-labs/windmill/commit/776e97875218476a46bfb72f8648b83471e70cc4)

Thank you so much for fixing this so quickly!

**Comment by marcelklehr** [comment]
Hello :wave: 
Another issue with origin URLs: When retrieving the resume URL in a script via `await wmill.getResumeUrls()` it does not seem to use the configured origin of windmill but uses localhost. Is this a config issue on our side or a bug?

**Comment by rubenfiszel** [comment]
Hello, that one is on your side. You need to set "Base Url" correctly in your instance settings

**Comment by bigcat88** [comment]
Good time of day. We have switched from the `v1.376.1` version to the `v1.391.0` and noticed two small bugs.

1. Clicking on different workspace leads to the double url (`/index.php/apps/app_api/proxy/flowhttp://nextcloud.local/index.php/apps/app_api/proxy/flow/`)

<img width="403" alt="image" src="https://github.com/user-attachments/assets/fae6afca-3e38-4bb5-bb03-fc2c98e57d49">

---

2. All left menu buttons work, except "Runs"

<img width="124" alt="image" src="https://github.com/user-attachments/assets/6a9c5029-f5f5-4f14-85bd-4abf038cea2a">

Looks like in new version prepending of VITE_URL is missing for "Runs", in console.log I see: Failed to load resource: the server responded with a status of 404 (Not Found) `http://nextcloud.local/runs?show_schedules=true&show_future_jobs=true`

**Comment by rubenfiszel** [comment]
@bigcat88 Fixed in 05a659034

**Comment by bigcat88** [comment]
Thanks a lot for the fixing.

Should I continue posting here about other places I found where the URLs are incorrect or create a new issue about them?

I really think that there aren't many places like this left, everything works where I click, but still, what would be more convenient for you?

Like for example clicking on "View script" here leads to `http://nextcloud.local/scripts/get/a821831a616a34bd`:

<img width="1256" alt="image" src="https://github.com/user-attachments/assets/9890b082-6ba3-48ec-b4e1-dc5378fe8f54">
